File: scrapers/extractedimage/mia.pl
{
	searchURL => 'http://www.artsmia.org/viewer/index.php?&v=2&dept[]=13&dept[]=1&class=print&start=%s',
	searchStart => 1,
	searchEnd => 3061,
	searchIncrement => 12,

	urlExtractor => scraper {
		process '//div[@class="result_label"]/a[contains(@href,"detail")]', 'urls[]' => '@href';	
	},

	cleanURL => sub {
		return "http://www.artsmia.org/viewer/" . $_[0];
	},

	extractPageID => qr/id=(\d+)/,

	pageExtractor => scraper {
		process '//span[@class="label"][text()="Title:"]/following-sibling::span[1]', title => 'TEXT';
		process '//span[@class="label"][text()="Artist:"]/following-sibling::span[1]', artist => 'TEXT';
		process '//span[@class="label"][text()="Date:"]/following-sibling::span[1]', date => 'TEXT';
		process "#g_image img", source_image => '@src';
	},

	cleanPage => sub {
		my $item = shift;
		$item->{'source_image'} =~ s/\d+\/medium/large/;
		return $item;
	},

	extractImageFile => qr/images\/(\d+)/
}

File: scrapers/extractedimage/mia.js
exports.extract = {
	title: [ '//span[@class="label"][text()="Title:"]/following-sibling::span[1]', title => title.replace( /_/g, "" ) ],
	artist: [ '//span[@class="label"][text()="Artist:"]/following-sibling::span[1]/text()[1]', artist => artist.replace( /\s*et\s*al\.?/, "" ).replace( /attributed to/, "" )
        .replace( /^(.*\s)(I+)(\s.*)$/, "$1$3 $2" ) ],
	date: '//span[@class="label"][text()="Date:"]/following-sibling::span[1]',
	description: [ '#primary_text || //span[@class="label"][text()="Physical Description:"]/following-sibling::span[1]', desc => desc.replace( /_/g, "" ) ],
	source_image: [ "#g_image//img/@src", name => name.replace( /\d+\/medium/, "large" ) ]
};

exports.reverseName = true;

exports.genURL = id => "http://www.artsmia.org/viewer/detail.php?id=" + id + "&v=2&dept[]=13&dept[]=1&class=print";

exports.getImage = name => /images\/(\d+)/.exec( name ) ?
    RegExp.$1 + ".jpg" :
    null;
